recipes-graphics/libjxl/libjxl.inc

```diff
@@ -0,0 +1,44 @@
+SUMMARY = "JPEG XL reference implementation"
+HOMEPAGE = "https://github.com/libjxl/libjxl/"
+BUGTRACKER = "https://github.com/libjxl/libjxl/issues"
+
+FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
+
+S = "${WORKDIR}/git"
+
+SRC_URI = "git://github.com/libjxl/libjxl.git;branch=v0.6.x"
+SRCREV = "a205468bc5d3a353fb15dae2398a101dff52f2d3"
+
+LICENSE = "BSD-3-Clause"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=6a905a337cc228a1f68f0b5416f52a7f"
+DEPENDS = "brotli highway"
+
+EXTRA_OECMAKE = " -DBUILD_TESTING=OFF \
+    -DJPEGXL_ENABLE_FUZZERS=OFF \
+    -DJPEGXL_ENABLE_DEVTOOLS=OFF \
+    -DJPEGXL_ENABLE_TOOLS=OFF \
+    -DJPEGXL_ENABLE_MANPAGES=OFF \
+    -DJPEGXL_ENABLE_BENCHMARK=OFF \
+    -DJPEGXL_ENABLE_EXAMPLES=OFF \
+    -DJPEGXL_ENABLE_JNI=OFF \
+    -DJPEGXL_ENABLE_VIEWERS=OFF \
+    -DJPEGXL_ENABLE_TCMALLOC=OFF \
+    -DJPEGXL_ENABLE_PLUGINS=OFF \
+    -DJPEGXL_ENABLE_COVERAGE=OFF \
+    -DJPEGXL_ENABLE_PROFILER=OFF \
+    -DJPEGXL_ENABLE_TRANSCODE_JPEG=OFF \
+    -DJPEGXL_ENABLE_SJPEG=OFF \
+    -DJPEGXL_STATIC=OFF \
+    -DJPEGXL_WARNINGS_AS_ERRORS=OFF \
+    -DJPEGXL_ENABLE_SKCMS=ON \
+    -DJPEGXL_FORCE_SYSTEM_BROTLI=ON \
+    -DJPEGXL_FORCE_SYSTEM_GTEST=ON \
+    -DJPEGXL_FORCE_SYSTEM_HWY=ON"
+
+inherit cmake
+
+CXXFLAGS:append:arm = " -mfp16-format=ieee"
+
+do_configure:prepend() {
+    ${S}/deps.sh
```

Comment:
Won't this download sources? AFAIU this is an antipattern, because it prevents doing full source downloads ahead of time, running the build steps non-networked, and also skips doing validation of the downloaded source files and their licenses.

Instead, you should add _also_ list the additional sources in `SRC_URI` above and add checksums for them. Like I did [in the build recipe for WebKit's Flatpak SDK](https://github.com/WebKit/WebKit/commit/0d87a259a958f39a4f4ddfd594c73bfc56e3dd33#diff-25da08c75484274613f2ddf7d04be801a247060957524fc6728452b4c360fab2R34).

Reply:
Honestly speaking yes, Yes, it is. Let me work on this.

Reply:
Let me know if now looks better.